actions/src/issues/checkIssueLabels.js

```diff
@@ -0,0 +1,129 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview File to handle checks when a issue is labeled.
+ */
+
+const core = require('@actions/core');
+const { context, GitHub } = require('@actions/github');
+const whitelist = require('../../../userWhitelist.json');
+const GOOD_FIRST_LABEL = 'good first issue';
+const prLabels = ['dependencies', 'critical', 'stale'];
+
+const checkLabels = async () => {
+  core.info('Checking newly added label...');
+  const token = core.getInput('repo-token');
+  const label = context.payload.label;
+  const octokit = new GitHub(token);
+  const user = context.payload.sender.login;
+
+  if (
+      label.name === GOOD_FIRST_LABEL &&
+      !whitelist.goodFirstIssue.includes(user)) {
+    core.info('Good first issue label got added by non whitelisted user.');
+    await handleGoodFirstIssueLabel(octokit, user);
+  } else if (prLabels.includes(label.name) || label.name.startsWith('PR')) {
+    core.info('PR label got added on an issue');
+    await handlePRLabel(octokit, label.name, user);
+  }
+};
+
+/**
+ * Handles cases when a good first issue gets added by a non whitelisted user.
+ *
+ * @param {import('@actions/github').GitHub} octokit
+ * @param {String} user - Username of the user that added the label.
+ */
+const handleGoodFirstIssueLabel = async (octokit, user) => {
+  const issueNumber = context.payload.issue.number;
+  // Comment on the issue and ping the onboarding team lead.
+  await octokit.issues.createComment(
+    {
+      body:'Hi @' + user + ', thanks for proposing this as a good first ' +
+          'issue. Looping in @' + whitelist.teamLeads.onboardingTeam +
+          ' to confirm, removing the label until it is approved.',
```

Comment:
Let us add full sentences: `I am removing the label for now and looping in @x to approve the label. It will be added back if approved. Thanks!`